interactor-impl/src/main/java/com/gtxtreme/template/interactor/content/GetContentInteractorImpl.kt

```diff
@@ -0,0 +1,34 @@
+package com.gtxtreme.template.interactor.content
+
+import com.gtxtreme.template.domain.base.Result
+import com.gtxtreme.template.domain.content.Content
+import com.gtxtreme.template.domain.content.GetContentUseCase
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableSharedFlow
+import timber.log.Timber
+
+class GetContentInteractorImpl(val getContentUseCase: GetContentUseCase) : GetContentInteractor {
+
+    private val searchContentResultStateFlow = MutableSharedFlow<List<Content>>()
+
+    override val searchResults: Flow<List<Content>>
+        get() = searchContentResultStateFlow
+
+    override suspend fun search(param: String) {
```

Comment:
Since we will only provide the coroutine scope in the view model while calling this function, this is where we should switch the execution to some other dispatcher.
```suggestion
    override suspend fun search(param: String) = coroutineContextController.switchToDefault {
```